crates/web-client/src/import.rs

```diff
@@ -69,6 +73,33 @@ impl WebClient {
         Ok(Account::from(generated_acct))
     }
 
+    #[wasm_bindgen(js_name = "importAccountById")]
+    pub async fn import_account_by_id(&mut self, account_id: String) -> Result<Account, JsValue> {
+        let client = self
+            .get_mut_inner()
+            .ok_or_else(|| JsValue::from_str("Client not initialized"))?;
+
+        let account_id = AccountId::from_hex(&account_id)
+            .map_err(|e| JsValue::from_str(&format!("invalid account id: {}", e)))?;
+
+        client
+            .import_account_by_id(account_id.clone())
+            .await
+            .map_err(|err| js_error_with_context(err, "failed to import public account"))?;
+
+        let record_opt = client
+            .get_account(account_id)
+            .await
+            .map_err(|err| js_error_with_context(err, "failed to retrieve account"))?;
+
+        if let Some(record) = record_opt {
+            let inner_acct = record.account().clone();
+            Ok(Account::from(inner_acct))
+        } else {
+            Err(JsValue::from_str("failed to parse account record"))
```

Comment:
nit: Maybe this is more precise
```suggestion
            Err(JsValue::from_str("account importing did not fail but the account was not found in the store"))
```

Reply:
I simplified the exported `import_account_by_id` function, it doesn't need these messages anymore.